### ISSUE TITLE
Local-install with non-root tsuser

### ISSUE THREAD

**Issue by havrazO** [Thinstation/thinstation#245]
On VM environment 
I use pxe tiny image with root user  to boot the client with it . then with ssh I install a syslinux non-root user image with local-install.sh
After that client boots from hdd load everything but then only black screen. 
How to make a non-root syslinux image updates it self automatically ?

**Comment by Doncuppjr** [comment]
I would set a cron job(runs as root), to install the update in the background, and then when the station reboots, it's all updated.
 

    On Tuesday, September 26, 2017 7:15 AM, Havraz Othman <notifications@github.com> wrote:
 

 On VM environment
I use pxe tiny image with root user to boot the client with it . then with ssh I install a syslinux non-root user image with local-install.sh
After that client boots from hdd load everything but then only black screen.
How to make a non-root syslinux image updates it self automatically ?—

**Comment by havrazO** [comment]
thx for the quick respond,

I tried to set the cron job in - thinstation.network,conf on the webserver didnt worked 
als in the syslinux image buildtime conf didnt work eather
my cronjob look like this,not sure if its right 
CRON_JOB='@reboot' root /bin/local-install.sh upgrade'
also tried 
CRON_JOB='@reboot' root local-install.sh upgrade'
CRON_JOB='@reboot' root /sbin/local-install.sh upgrade'
CRON_JOB='@reboot' user /bin/local-install.sh upgrade'

after station boots from hdd it can't get new configs even if i increment Version nummber
i dont know what i doing wrong 

- net file enabled is on in buildtime conf
- the local-install package is included 
- package switchuser is included 
- package userauto is included
- param tsuser is set
- rootpasswd is set
- hdupdates package is inculded and it works 

bootverbosity set to 32 and i get a massage at boot 
/etc/rc2.d/s01local-install_mount: line 1: /bin/local-install.sh: Permission denied

**Comment by havrazO** [comment]
i found a solution which worked for me 
first i made my pxe image a user image param tsuser= e.g "goofy"
also before build i added to this line 

https://github.com/Thinstation/thinstation/blob/8790531eaf38c71ad30f9c15f4001e8629e4cf03/ts/build/packages/local-install/bin/local-install.sh#L144

`chown -R goofy <goofy home directory>`

an i build my syslinux image also as user goofy  the image which will be permanent on the hdd and get auto upgrade

so after i local-install with pxe image the syslinux image boots without errors/blackscreen

**Comment by MarcusEyre** [comment]
Just a note about the cron-job. local-install will run the upgrade during boot but I see you run it during shutdown which is probably better since the it will not slow down the boot process if it has already been updated during shutdown.

/Marcus

**Comment by havrazO** [comment]
hey Marcus,
I learned alot "and still learning" about this great project , I had the problem with getting tsuser image in my test-environment and I got it solved by changing ownership to the tsuser.. as above,and changed my tiny installation image to and tsuser image as well with the same tsuser name so the user home directory will be creating during the first install.
i also found out that x11vnc wasn't working, so i added Autoruns package to build and in buildtime added  "changing ownerships cmds" I think now  corn-job is working well because the check_x11vnc.sh also works in the background. 
maybe there is still a better way to make a cleaner tsuser image which runs all that without changing ownerships in code.
